File: src/main/java/org/tindalos/principle/domain/checker/PackageAnalyzer.java
package org.tindalos.principle.domain.checker;

import java.util.List;

import org.tindalos.principle.domain.core.Package;
import org.tindalos.principle.domain.core.DesignQualityCheckConfiguration;

public interface PackageAnalyzer {
    
    List<Package> analyze(DesignQualityCheckConfiguration parameters);

}

File: src/main/java/org/tindalos/principle/domain/checker/PackageAnalyzer.scala
package org.tindalos.principle.domain.checker

import org.tindalos.principle.domain.core.DesignQualityCheckConfiguration
import org.tindalos.principle.domain.core.Package

trait PackageAnalyzer {
  
  def analyze(configuration: DesignQualityCheckConfiguration): List[Package]

}